packages/apps/plugins/plugin-space/src/SpacePlugin.tsx

```diff
@@ -1033,28 +1018,96 @@ export const SpacePlugin = ({
+
+              if (!intent.undo) {
+                // Capture the current state for undo
+                const deletionData = {
+                  object,
+                  parentCollection,
+                  index:
+                    parentCollection instanceof CollectionType
+                      ? parentCollection.objects.indexOf(object as Expando)
+                      : -1,
+                  nestedObjects: object instanceof CollectionType ? [...object.objects] : [],
+                  wasActive: isIdActive(navigationPlugin?.provides.location.active, objectId),
+                };
+
+                // If the item is active, navigate to "nowhere" to avoid navigating to a removed item.
+                if (deletionData.wasActive) {
+                  await intentPlugin?.provides.intent.dispatch({
+                    action: NavigationAction.CLOSE,
+                    data: { activeParts: { main: [objectId], sidebar: [objectId], complementary: [objectId] } },
+                  });
+                }
+
+                if (parentCollection instanceof CollectionType) {
+                  // TODO(Zan): Is there a nicer way to do this without casting to Expando?
+                  const index = parentCollection.objects.indexOf(object as Expando);
+                  if (index !== -1) {
+                    parentCollection.objects.splice(index, 1);
+                  }
+                }
+
+                // If the object is a collection, move the objects inside of it to the collection above it.
+                if (object instanceof CollectionType && parentCollection instanceof CollectionType) {
+                  object.objects.forEach((obj) => {
+                    if (!parentCollection.objects.includes(obj)) {
+                      parentCollection.objects.push(obj);
+                    }
+                  });
+                }
+
+                space.db.remove(object as any);
+
+                const undoMessageKey =
+                  object instanceof CollectionType ? 'collection deleted label' : 'object deleted label';
+
                 return {
-                  intents: [
-                    [
-                      {
-                        action: LayoutAction.SET_LAYOUT,
-                        data: {
-                          element: 'popover',
-                          anchorId: `dxos.org/ui/${caller}/${fullyQualifiedId(object)}`,
-                          component: 'dxos.org/plugin/space/RemoveObjectPopover',
-                          subject: {
-                            object,
-                            collection: intent.data?.collection,
-                          },
-                        },
-                      },
-                    ],
-                  ],
+                  data: true,
+                  undoable: {
+                    message: translations[0]['en-US'][SPACE_PLUGIN][undoMessageKey], // Consider using a translation key here
```

Comment:
we're basically limited to this type of usage until we expose the translation utils outside of react. It is a known issue, but not one that we've encountered enough to prioritize fixing it

Reply:
Oh, that comment is vestigial.